src/pfe/file-watcher/server/src/projects/projectSpecifications.ts

```diff
@@ -453,6 +453,81 @@ const reconfigIgnoredFilesForDaemon = async function (ignoredPaths: string[], op
     io.emitOnListener("projectSettingsChanged", data);
 };
 
+/**
+ * @function
+ * @description Reconfig the ref paths for a project.
+ *
+ * @param args <Required | Any> - Required arguments for reconfiguration of ref paths.
+ *
+ * @returns Promise<any>
+ */
+const reconfigRefPathsForDaemon = async function (refPaths: RefPath[], operation: Operation): Promise<any> {
+    const projectInfo: ProjectInfo = operation.projectInfo;
+    const projectID = projectInfo.projectID;
+
+    // if defined, do some checks, sanitized the input etc
+    if (refPaths) {
```

Comment:
Do we need this check?
If it's undefined, it would be filtered out in `projectSpecificationHandler`:
https://github.com/eclipse/codewind/blob/d0ec065f591c9396b08045416aacc9c00e82d03a/src/pfe/file-watcher/server/src/projects/projectSpecifications.ts#L972-L981

Reply:
Nevermind for this one. Saw the check for `refPaths` removal.

Reply:
The check is needed because the function could be passed an undefined `refPaths` for the removal case

https://github.com/eclipse/codewind/pull/1535/files#diff-2a30a2a4dccd9b8e1dccf2220b27057fR992